oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/OAuth2Tokens.java

```diff
@@ -87,19 +89,15 @@ public OAuth2RefreshToken getRefreshToken() {
 	/**
 	 * Returns the token specified by {@code token}.
 	 *
-	 * @param token the token
-	 * @param <T> the type of the token
-	 * @return the token, or {@code null} if not available
+	 * @param token the token value
+	 * @return an optional wrapped {@code AbstractOAuth2Token}
 	 */
-	@Nullable
-	@SuppressWarnings("unchecked")
-	public <T extends AbstractOAuth2Token> T getToken(String token) {
+	public Optional<AbstractOAuth2Token> getToken(String token) {
```

Comment:
Please revert all changes in `OAuth2Tokens`.
FYI, this will be removed and code merged into `OAuth2Authorization` as part of a remodelling exercise.